spec/lib/whitehall_importer/create_file_attachment_revision_spec.rb

```diff
@@ -4,26 +4,42 @@
   let(:whitehall_file_attachment) do
     build(:whitehall_export_file_attachment)
   end
+  let(:document_import) { build(:whitehall_migration_document_import) }
 
   context "creates a file attachment" do
     it "fetches file from asset-manager" do
-      create_revision = described_class.new(whitehall_file_attachment)
+      create_revision = described_class.new(document_import, whitehall_file_attachment)
       expect(create_revision.call).to have_requested(:get, whitehall_file_attachment["url"])
     end
 
     it "creates a FileAttachment::Revision and sets correct metadata" do
       revision = nil
-      expect { revision = described_class.call(whitehall_file_attachment) }
+      expect { revision = described_class.call(document_import, whitehall_file_attachment) }
         .to change { FileAttachment::Revision.count }.by(1)
 
       expect(revision.metadata_revision.title).to eq(whitehall_file_attachment["title"])
       expect(revision.filename).to eq("some-txt.txt")
     end
+
+    it "should create a WhitehallMigration::AssetImport for each attachment variant" do
+      asset_import = WhitehallMigration::AssetImport
+      revision = nil
+      expect { revision = described_class.call(document_import, whitehall_file_attachment) }
+        .to change { asset_import.where(variant: nil).count }.by(1)
+        .and change { asset_import.where(variant: "thumbnail").count }.by(1)
+        .and change { asset_import.count }.by(2)
+      expect(asset_import.find_by(variant: nil).file_attachment_revision).to eq(revision)
+      expect(asset_import.find_by(variant: "thumbnail").file_attachment_revision).to eq(revision)
+      expect(asset_import.find_by(variant: nil).original_asset_url)
+        .to eq(whitehall_file_attachment["url"])
+      expect(asset_import.find_by(variant: "thumbnail").original_asset_url)
+        .to eq(whitehall_file_attachment["variants"]["thumbnail"]["url"])
```

Comment:
We've hit quite a lot of assertions in this test I wonder if we can reduce them to make the test more focused. Maybe something like:

```
revision = described_class.call(document_import, whitehall_file_attachment)

expect(document_import.assets.size).to eq(2)
expect(document_import.assets.map(&:attributes)).to contain_exactly(
  a_hash_including(variant: nil,
                   file_attachment_revision_id: revision.id,
                   original_asset_url: whitehall_file_attachment["url"]),
  a_hash_including(variant: "thumbnail,
                   file_attachment_revision_id: revision.id,
                   original_asset_url: whitehall_file_attachment["variants"]["thumbnail"]["url"]),
)
```